src/jit/codegenarm.cpp

```diff
@@ -308,7 +308,11 @@ void CodeGen::genReturn(GenTreePtr treeNode)
             genConsumeReg(op1);
 
             regNumber retReg = varTypeIsFloating(treeNode) ? REG_FLOATRET : REG_INTRET;
-            if (op1->gtRegNum != retReg)
+            if (varTypeIsFloating(treeNode) && compiler->opts.compUseSoftFP)
+            {
+                getEmitter()->emitIns_R_R(INS_vmov_f2i, EA_4BYTE, REG_INTRET, op1->gtRegNum);
+            }
+            else if (op1->gtRegNum != retReg)
             {
```

Comment:
It seems like this might not be sufficient. E.g., registerfp.cpp for the `LEGACY_BACKEND` case has:
```
            if (compiler->info.compIsVarArgs || compiler->opts.compUseSoftFP)
            {
                if (tree->TypeGet() == TYP_FLOAT)
                {
                    inst_RV_RV(INS_vmov_f2i, REG_INTRET, REG_FLOATRET, TYP_FLOAT, EA_4BYTE);
                }
                else
                {
                    assert(tree->TypeGet() == TYP_DOUBLE);
                    inst_RV_RV_RV(INS_vmov_d2i, REG_INTRET, REG_NEXT(REG_INTRET), REG_FLOATRET, EA_8BYTE);
                }
            }
```
so, it handles both float and double, and also considers the varargs case (which I guess we don't have for .NET Core, but doesn't hurt to check anyway, in case we need to implement it at some point)